packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -135,14 +136,27 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
               shouldRequestFocus: false,
               supportClearField: true,
               onSelection: _onSelection,
-              tracking: true,
               decoration: const InputDecoration(
                 contentPadding: EdgeInsets.all(denseSpacing),
                 border: OutlineInputBorder(),
                 focusedBorder: OutlineInputBorder(borderSide: BorderSide.none),
                 enabledBorder: OutlineInputBorder(borderSide: BorderSide.none),
                 labelText: 'Eval',
               ),
+              determineOverlayXPosition:
+                  (String inputValue, TextStyle inputStyle) {
+                // X-coordinate is equivalent to the width of the input text
+                // up to the last "." or the insertion point (cursor):
+                final textSegment = inputValue.contains('.')
```

Comment:
nit: slightly cleaner is:
```dart
final indexOfDot = inputValue.lastIndexOf('.')
final textSegment = indexOfDot != -1 ? inputValue.substring(0, indexOfDot + 1) : inputValue;
```
A little better as it doesn't have two different expressions with the same goal (contains and lastIndexOf) and is a little faster due to less dupe work not that it matters in this particular case.